addons/payment/models/payment_transaction.py

```diff
@@ -1129,4 +1129,6 @@ def _get_last(self):
         :return: The last transaction of the recordset, sorted by id.
         :rtype: recordset of `payment.transaction`
         """
-        return self.filtered(lambda t: t.state != 'draft').sorted()[:1]
+        if self:
+            return self.filtered(lambda t: t.state != 'draft').sorted()[:1]
+        return self.env['payment.transaction']
```

Comment:
In a call to _get_last the recordset might be empty (partner is then set to false) and access rules will fail.